src/pages/settings/Profile/LoginField.js

```diff
@@ -82,46 +71,52 @@ class LoginField extends Component {
                 note = this.props.translate('loginField.numberHasNotBeenValidated');
             }
 
-        // Has unvalidated email
+            // Has unvalidated email
         } else if (this.props.login.partnerUserID && !this.props.login.validatedDate) {
             note = this.props.translate('loginField.emailHasNotBeenValidated');
         }
 
         return (
-            <View style={styles.mb6}>
-                <Text style={[styles.formLabel, this.getLabelMargin()]}>{this.props.label}</Text>
-                {!this.props.login.partnerUserID ? (
-                    <View style={[styles.mln5, styles.mrn5]}>
-                        <MenuItem
-                            key={`common.add.${this.props.type}`}
-                            title={`${this.props.translate('common.add')} ${this.props.label}`}
-                            icon={Expensicons.Plus}
-                            onPress={() => Navigation.navigate(ROUTES.getSettingsAddLoginRoute(this.props.type))}
-                        />
-                    </View>
-                ) : (
-                    <View style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter]}>
-                        <Text numberOfLines={1}>
-                            {this.props.type === CONST.LOGIN_TYPE.PHONE
-                                ? this.props.toLocalPhone(this.props.login.partnerUserID)
-                                : this.props.login.partnerUserID}
-                        </Text>
-                        {!this.props.login.validatedDate && (
-                            <Button
-                                small
-                                style={[styles.mb2]}
-                                onPress={this.onResendClicked}
-                                ContentComponent={() => (this.state.showCheckmarkIcon ? (
-                                    <Icon fill={themeColors.inverse} src={Expensicons.Checkmark} />
-                                ) : (
-                                    <Text style={styles.buttonSmallText}>
-                                        {this.props.translate('common.resend')}
-                                    </Text>
-                                ))}
+            <View style={[styles.ph8]}>
+                <View>
+                    {!this.props.login.partnerUserID || this.props.login.validatedDate ? (
+                        <View style={[styles.mln8, styles.mrn8]}>
+                            <MenuItemWithTopDescription
+                                title={this.props.login.partnerUserID ? (this.props.type === CONST.LOGIN_TYPE.PHONE
+                                    ? this.props.toLocalPhone(this.props.login.partnerUserID)
+                                    : this.props.login.partnerUserID) : this.props.label}
```

Comment:
Seems you're gunna have to clean this up due to the Linter :D But either way I think it's much more readable if we move this logic before the JSX & maybe calculate the title in multiple steps

Reply:
fixed.